### PR TITLE
rename illumina related classes

### PR DESCRIPTION
**Pull request by ChrOertlin**
### Added

-

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/apps/demultiplex/sample_sheet/sample_models.py [hunk 1]
```diff
@@ -17,8 +17,8 @@
 LOG = logging.getLogger(__name__)
 
 
-class FlowCellSample(BaseModel):
-    """Class that represents a flow cell sample."""
+class IlluminaIndexSettings(BaseModel):
```
Comment (henrikstranneheim) [inline cg/apps/demultiplex/sample_sheet/sample_models.py]: Should it not be singular, so it reads:
` IlluminaIndexSetting.lane`

as you access each setting individually (usually)

File: cg/apps/demultiplex/sample_sheet/read_sample_sheet.py [hunk 2]
```diff
@@ -2,14 +2,14 @@
...
-from cg.apps.demultiplex.sample_sheet.sample_models import FlowCellSample
+from cg.apps.demultiplex.sample_sheet.sample_models import IlluminaSampleIndexSettings
 from cg.constants.demultiplexing import SampleSheetBcl2FastqSections, SampleSheetBCLConvertSections
 from cg.exc import SampleSheetContentError, SampleSheetFormatError
 
 LOG = logging.getLogger(__name__)
 
 
-def validate_samples_are_unique(samples: list[FlowCellSample]) -> None:
+def validate_samples_are_unique(samples: list[IlluminaSampleIndexSettings]) -> None:
```
Comment (henrikstranneheim) [inline cg/apps/demultiplex/sample_sheet/read_sample_sheet.py:12]: Looks like a list with a list of IlluminaIndexSetting objects to me due to the plural.

**Review by henrikstranneheim — COMMENTED** [review]
🧹 👍

**Review by diitaz93 — APPROVED** [review]
Looks good 👍

File: cg/apps/demultiplex/sample_sheet/read_sample_sheet.py (hunk 2 above)
Comment (diitaz93) [reply cg/apps/demultiplex/sample_sheet/read_sample_sheet.py:12]: Other option would be `IlluminaSampleIndex`

**Pull request merged**